sklearn/decomposition/truncated_svd.py

```diff
@@ -160,7 +161,8 @@ def fit_transform(self, X, y=None):
         random_state = check_random_state(self.random_state)
 
         if self.algorithm == "arpack":
-            U, Sigma, VT = svds(X, k=self.n_components, tol=self.tol)
+            v0 = _init_arpack_v0(min(X.shape), self.random_state)
```

Comment:
This is going to change the results of the function.
1. Is it beneficial and worth breaking backwards compatibility?
2. If so, it needs to be documented in the change log.

Reply:
1. Without this initialization, the v0 is initialized with randn. In arpack it is initialized with uniform distrib on [-1, 1]. The goal is also to generalize the initialization of the calls to arpack lib are initialized and provide a random state.
2. What should be documented ? For each impacted algorithm : "Initialization of v0 is now uniform [-1, 1]" something of this flavour ?